src/libexpr/eval-cache.hh

```diff
@@ -44,6 +44,7 @@ enum AttrType {
     Misc = 4,
     Failed = 5,
     Bool = 6,
+    ListOfStrings = 7,
```

Comment:
Adding support for arbitrary lists would add a lot of complexity (cursors are based on attrpaths, and there is no obvious key corresponding to list elements).

I don't think a schema change is necessary, since older versions won't query these attributes.

Reply:
> I don't think a schema change is necessary, since older versions won't query these attributes.

I indeed couldn’t manage to break an old version, but it’s actually easy to fill the cache db with something that confuses the new version:

```console
# Clear the cache
$ rm ~/.cache/nix/eval-cache-v2/*
# Use current master to fill the cache with something that will confuse Nix from this branch
$ nix run nix -- eval nixpkgs#hello.meta.outputsToInstall
[ "out" ]
# And now `nix build` is broken
$ nix run github:edolstra/nix/respect-outputs-to-install -- build nixpkgs#hello
error: 'legacyPackages.x86_64-linux.hello.meta.outputsToInstall' is not a list of strings
```

Reply:
> cursors are based on attrpaths, and there is no obvious key corresponding to list elements

I recall there was one part of the old CLI where one could in fact do `foo.3`? I would not be opposed to such positional indexing on the command line.

Reply:
I've bumped the schema version to `v3`.